test/trento/application/usecases/hosts_test.exs

```diff
@@ -38,4 +41,70 @@ defmodule Trento.HostsTest do
       refute deregistered_host.id in hosts_ids
     end
   end
+
+  describe "deregister_host" do
+    test "should deregister a host with unknown health" do
```

Comment:
I'm totally disoriented with these tests.
As far as I see the code, I don't see any place where the heartbeat state is even used.
How does this work?
Please, have a look on the tests. I don't find any logic on this.

And maybe add `setup :verify_on_exit!` on top of the test file. I think almost all (if not all) the expects are wrong.